cpp/bench/ann/src/diskann/diskann_benchmark.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "../common/ann_types.hpp"
+#include "diskann_wrapper.h"
+#include "../common/conf.hpp"
+
+#define JSON_DIAGNOSTICS 1
+#include <nlohmann/json.hpp>
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+#include <utility>
+
+namespace cuvs::bench {
+
+template <typename T>
+void parse_build_param(const nlohmann::json& conf,
+                       typename cuvs::bench::diskann_memory<T>::build_param& param)
+{
+  param.R = conf.at("R");
+  if (conf.contains("L_build")) { param.L_build = conf.at("L_build"); }
+  if (conf.contains("alpha")) { param.num_threads = conf.at("alpha"); }
+  if (conf.contains("num_threads")) { param.num_threads = conf.at("num_threads"); }
+}
+
+template <typename T>
+void parse_build_param(const nlohmann::json& conf,
+                       typename cuvs::bench::diskann_ssd<T>::build_param& param)
+{
+  param.R = conf.at("R");
+  if (conf.contains("L_build")) { param.L_build = conf.at("L_build"); }
+  if (conf.contains("alpha")) { param.num_threads = conf.at("alpha"); }
+  if (conf.contains("num_threads")) { param.num_threads = conf.at("num_threads"); }
+  if (conf.contains("QD")) { param.QD = conf.at("QD"); }
+  param.dataset_base_file = cuvs::bench::configuration::singleton().get_dataset_conf().base_file;
+  for (auto index : cuvs::bench::configuration::singleton().get_indices()) {
+    if (index.build_param == param) {
+      param.index_file = index.file;
```

Comment:
Nitpick: stop iterating as soon as we find the item.
```suggestion
      param.index_file = index.file;
      break;
```